fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraAcl.java

```diff
@@ -225,4 +230,39 @@ public Response getResource(@HeaderParam("Range") final String rangeValue)
             readLock.release();
         }
     }
+
+    /**
+     * Deletes an object.
+     *
+     * @return response
+     */
+    @DELETE
+    @Timed
+    public Response deleteObject() {
+
+        hasRestrictedPath(externalPath);
+        LOGGER.info("Delete resource '{}'", externalPath);
+
+        final AcquiredLock lock = lockManager.lockForDelete(resource().getPath());
+
+        try {
+            final FedoraResource aclResource = resource().getAcl();
+            if (aclResource != null) {
+                aclResource.delete();
+            }
+            session.commit();
+
+            if (aclResource != null) {
+                return noContent().build();
+            } else {
+                return notFound();
```

Comment:
In keeping with the Jersey ExceptionMapper pattern, maybe throw a `ItemNotFoundException` instead of creating the response manually.